modin/core/dataframe/algebra/binary.py

```diff
@@ -230,13 +245,25 @@ def caller(
                         elif infer_dtypes == "float":
                             dtypes = compute_dtypes_common_cast(query_compiler, other)
                             dtypes = dtypes.apply(coerce_int_to_float64)
+                    if (
+                        query_compiler._modin_frame._columns_cache is not None
+                        and other._modin_frame._columns_cache is not None
+                    ):
+                        if (
+                            len(query_compiler.columns) == 1
+                            and len(other.columns) == 1
+                            and len(query_compiler.columns.difference(other.columns))
+                            == 0
```

Comment:
btw, why a nested if-statement here? can't we just put this whole construction into a single condition?
```suggestion
                    if (
                        query_compiler._modin_frame._columns_cache is not None
                        and other._modin_frame._columns_cache is not None
                        and len(query_compiler.columns) == 1
                        and len(other.columns) == 1
                        and query_compiler.columns.equals(other.columns)
                    ):
```
code execution guarantees that the rest of the condition will not be computed if any of the preceding ones is false, meaning that we are not going to execute `.columns` computation if the cache is not available

Reply:
looks a bit bulky to me, left as is.